src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs

```diff
@@ -272,6 +272,21 @@ public void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
             }
         }
 
+        public void ChangeDefaultCredentialsPolicy(SafeWinHttpHandle requestHandle, bool allowDefaultCredentials)
+        {
+            uint optionData = allowDefaultCredentials ?
+                Interop.WinHttp.WINHTTP_AUTOLOGON_SECURITY_LEVEL_LOW :
```

Comment:
In addition, there are times when we don't want to automatically send default creds. We should only do so if the developer has set CredentialCache.DefaultCredentials.  So, using MEDIUM (instead of HIGH for example) would send them if challenged by the proxy/server.  So, we always use an opt-in model.